docs/components/api.md

```diff
@@ -184,7 +184,7 @@ The following primitives can be imported from the `@stencil/core` package and us
 
 - **getElement()**: Retrieve a Stencil element for a given reference.
 
-  __Type:__ `(ref: getElement) => string`<br />
+  __Type:__ `(ref: any) => HTMLStencilElement`<br />
```

Comment:
Shouldn't it be rather
```suggestion
  __Type:__ `(ref: HTMLElement) => HTMLStencilElement`<br />
```

Reply:
I went with `any` to match the type in [`stencil-public-runtime.ts`][1]. That being said, I agree it'd be great to further narrow the type. I see `(ref: HTMLElement) => HTMLStencilElement` matches the example, but reading through the other docs and examples on the api page, it seems like a `ref` can refer to either an `HTMLElement` or a `ComponentInterface`.

For example, `forceUpdate` and `getMode` are typed on the doc page as `(ref: HTMLElement) => void`, but their example code shows passing in a component instance. Similarly, `setMode` is typed the same but its example show passing in an arrow function.

@christian-bromann, I'm happy to fix up the types and examples, but I need to know how to type `ref`. Is `ComponentInterface | HTMLElement` sufficient? Would you anticipate any problems with reflecting this type back to the https://github.com/ionic-team/stencil repo?

[1]: https://github.com/ionic-team/stencil/blob/c3d4e8b170b405ef420236f12d7b19e21e541a81/src/declarations/stencil-public-runtime.ts#L337

Reply:
I think the reason why people put `any` in there in the first place was because the actual real type is not suppose to be exposed for public consumption. Hence, let's keep it close to how it is currently in the Stencil public documentation and disregard my comment here.